runtimes/huggingface/mlserver_huggingface/runtime.py

```diff
@@ -80,25 +77,23 @@ async def load(self) -> bool:
         return self.ready
 
     async def predict(self, payload: InferenceRequest) -> InferenceResponse:
-        """
-        TODO
-        """
-
-        # TODO: convert and validate?
-        kwargs = self.decode_request(payload, default_codec=MultiStringRequestCodec)
 
+        kwargs = self.decode_request(payload, default_codec=HuggingfaceRequestCodec)
         args = kwargs.pop("args", [])
-        X = kwargs.pop("array_inputs", None)
-        if X is not None:
-            args = [list(X)] + args
+
         prediction = self._model(*args, **kwargs)
+        return self.encode_response(prediction, default_codec=HuggingfaceRequestCodec)
+
+    async def metadata(self) -> MetadataModelResponse:
 
-        # TODO: Convert hf output to v2 protocol, for now we use to_json
-        str_out = [json.dumps(pred, cls=NumpyEncoder) for pred in prediction]
-        prediction_encoded = StringCodec.encode_output(payload=str_out, name="output")
+        task = self.hf_settings.task
+        metadata = metainfo_for(task)
 
-        return InferenceResponse(
-            model_name=self.name,
-            model_version=self.version,
-            outputs=[prediction_encoded],
+        model_metadata = MetadataModelResponse(
+            name=self.name,
+            platform=self._settings.platform,
+            versions=self._settings.versions,
+            inputs=self._settings.inputs or metadata.get("inputs", []),
+            outputs=self._settings.outputs or metadata.get("outputs", []),
         )
+        return model_metadata
```

Comment:
Oh this is brilliant! It's gonna be so useful to expose the pipeline's metadata here! 

Having said that, the best way to do this would be to edit the `self.inputs` and / or `self.outputs` fields, similar to how the MLflow's runtime handles these changes:

https://github.com/SeldonIO/MLServer/blob/f5297d42f00be647954bbed8b12b0bea29c00994/runtimes/mlflow/mlserver_mlflow/runtime.py#L132-L161